libmamba/src/download/mirror_impl.cpp

```diff
@@ -71,7 +71,7 @@ namespace mamba::download
      * OCIMirror implementation *
      ****************************/
 
-    namespace
+    namespace utils
```

Comment:
Why turning it into a non anonymous namespace? Did you have conflict with functions defined in another namespace?

Reply:
To have access in the [tests](https://github.com/mamba-org/mamba/pull/3310/files#diff-409867cced670ed6abe27e11cb1d7c4a1a4363d90b094eb5f1dd3016bc5ff0f9R15-R54).

Reply:
In that case I think it would be worth having a preprocessor token defining the namespace to util if we build the test, and emtpy if not. Otherwise this symbol (and the other ones included in this namespace) will be exported and visible from other libraries, which is something we wanted to avoid with the anonymous namespace.

@Klaim any thought on this?

Reply:
You are correct that naming the scope makes the symbols visible, anonymous namespaces are essentially similar to marking everything `static` (and more). So this is not a small change.

For testing internal functions there are various usual strategies:
- making the internal functions visible (like here);
- tests link with a "utility library" version of the library, which is essentially like a static library except it's not removing unused functions at linking;
- real proper "unit tests" as in test source file which is specific to that component and runs as a single executable, testing only that sub-component;
- the tests "include" the cpp they test (which can lead to ODR violations, depends on how both are coded);

From the current mamba organisation, looks like the first solution is the easiest to setup at the moment. Alternating the definition of the namespace might not work properly because this is an internal translation unit of libmamba, so you would have to build libmamba twice, once with the visible symbols, once with them hidden.

Reply:
Ok then let's keep it as is for now, if it clashes with other symbols when linking, we will simply change its name.